File: iOS/Code/Xamarin/bugTrap/bugTrapKit/Annotate/PathAnnotation.cs
using System;
using UIKit;

namespace bugTrapKit
{
	public class PathAnnotation : IAnnotationData
	{
		public UIBezierPath Path { get; private set; }

		#region IAnnotationData implementation

		public UIColor Color { get; private set; }

		public Annotate.Tool Tool { get; private set; }

		#endregion

		public PathAnnotation (Annotate.Tool tool, UIBezierPath path, UIColor color)
		{
			Tool = tool;

			var bzPath = UIBezierPath.FromPath(path.CGPath);
			bzPath.LineWidth = path.LineWidth;

			Path = bzPath;

			Color = color;
		}
	}
}

File: iOS/Code/Swift/bugTrap/bugTrapKit/Annotate/PathAnnotation.swift
//
//  PathAnnotation.swift
//  bugTrap
//
//  Created by Colby L Williams on 8/31/14.
//  Copyright (c) 2014 bugTrap. All rights reserved.
//

import UIKit

struct PathAnnotation : AnnotationData {

	let color: UIColor

	let	path: UIBezierPath

	let tool: Annotate.Tool
	
	// init (tool: Annotate.Tool, path: UIBezierPath, scale: CGFloat, color: UIColor) {
	init (tool: Annotate.Tool, path: UIBezierPath, color: UIColor) {
	
		self.tool = tool
		
		let bzPath = UIBezierPath(CGPath: path.CGPath)
			bzPath.lineWidth = path.lineWidth
			// bzPath.applyTransform(CGAffineTransformMakeScale(scale, scale))
		
		self.path = bzPath
	
		self.color = color
	}
}